### PR TITLE
 P1186R3 When do you actually use <=>?

### LINKED ISSUE #2985: [2019-07 CWG Motion 6] P1186R3 When do you actually use <=>?

### PR DESCRIPTION
**Pull request by zygoloid**
Fixes #2985

### CODE REVIEW

File: source/classes.tex [hunk 1]
```diff
@@ -6785,10 +6785,76 @@
 \rSec2[class.spaceship]{Three-way comparison}
 \indextext{operator!three-way comparison!defaulted}%
 
+\pnum
+The synthesized three-way comparison
```
Comment (burblebee) [inline source/classes.tex]: Please add the index entry `\indextext{operator!three-way comparison!synthesized}`.  FWIW, I would like to see "synthesized three-way comparison" kept as a definition as suggested in the paper - I realize we don't tend to do that when the term is only used locally, but what happens when it starts getting used elsewhere?

File: source/classes.tex [hunk 2]
```diff
@@ -6785,10 +6785,76 @@
...
+a == b ? partial_ordering::equivalent :
+a < b  ? partial_ordering::less :
+b < a  ? partial_ordering::greater :
+         partial_ordering::unordered
+\end{codeblock}
+
+\item
+Otherwise, if \tcode{R} is \tcode{strong_equality}, then
+\begin{codeblock}
+a == b ? strong_equality::equal : strong_equality::nonequal
```
Comment (burblebee) [inline source/classes.tex]: Suggestion: place `strong_equality::nonequal;` on new line for consistency with the previous items?

File: source/classes.tex [hunk 3]
```diff
@@ -6785,10 +6785,76 @@
...
+\end{codeblock}
+
+\item
+Otherwise, if \tcode{R} is \tcode{weak_equality}, then
+\begin{codeblock}
+a == b ? weak_equality::equivalent : weak_equality::nonequivalent
```
Comment (burblebee) [inline source/classes.tex]: ditto?

File: source/classes.tex [hunk 4]
```diff
@@ -6785,10 +6785,76 @@
...
+\pnum
+The synthesized three-way comparison
+for comparison category type \tcode{R}\iref{cmp.categories}
+of glvalues \tcode{a} and \tcode{b} of the same type
+is defined as follows:
+
+\begin{itemize}
+\item
+If overload resolution for \tcode{a <=> b}
+finds a usable function \iref{over.match},
```
Comment (burblebee) [inline source/classes.tex]: Nitpik: Please remove preceding space before `\iref` for consistency.

File: source/classes.tex (hunk 2 above)
Comment (zygoloid) [reply source/classes.tex]: Tried it, it looked a bit weird. We often only flow things onto multiple lines if they don't fit on one line; I think this case is similar.

File: source/classes.tex (hunk 4 above)
Comment (zygoloid) [reply source/classes.tex]: Done.

File: source/classes.tex (hunk 1 above)
Comment (zygoloid) [reply source/classes.tex]: Done.

**Pull request merged**